src/encoders/NvMediaH264Encoder.cpp

```diff
@@ -51,8 +53,8 @@ void NvMediaH264Encoder::set_encode_init_params()
   encodeInitParams_.encodeHeight = static_cast<uint16_t>(height_);
   encodeInitParams_.frameRateNum = framerate_;
   encodeInitParams_.frameRateDen = 1;
-  encodeInitParams_.profile = 0;
-  encodeInitParams_.level = 0;
+  encodeInitParams_.profile = profile_;
```

Comment:
Wouldn't it have been easier to have forced typecast here, rather than making NvMediaH264Encoder() outer interface (constructor) be dependent on Nvidia-specific data types? Would have allowed keeping nvmedia_common_encode.h include in cpp file rather than in header, would speed up compilation.

And why this typecast did not happen automatically in previous code, as it seems to be in case of `profile_ = profile` in CameraH264() constructor?